spot-contracts/contracts/FeePolicy.sol

```diff
@@ -306,13 +310,21 @@ contract FeePolicy is IFeePolicy, OwnableUpgradeable {
     }
 
     /// @inheritdoc IFeePolicy
-    function computeDeviationRatio(IFeePolicy.SubscriptionParams memory s) external view returns (uint256) {
+    function computeDeviationRatio(IFeePolicy.SubscriptionParams memory s) public view returns (uint256) {
+        return computeDeviationRatio(s.perpTVL, s.vaultTVL, s.seniorTR);
+    }
+
+    /// @inheritdoc IFeePolicy
+    function computeDeviationRatio(
+        uint256 perpTVL,
+        uint256 vaultTVL,
+        uint256 seniorTR
+    ) public view returns (uint256) {
         // NOTE: We assume that perp's TVL and vault's TVL values have the same base denomination.
         return
-            s.vaultTVL.mulDiv(s.perpTR, (s.perpTVL * s.vaultTR), MathUpgradeable.Rounding.Up).mulDiv(
+            vaultTVL.mulDiv(ONE, targetSubscriptionRatio).mulDiv(
```

Comment:
Does this revert to the earlier, more complicated computation?

Reply:
It is essentially the same, we just calculate `vaultTR` (or the juniorTR) instead of passing it down.

Reply:
This would be the right version
```
return
            (vaultTVL*ONE)                
                .mulDiv(seniorTR, (perpTVL * (TRANCHE_RATIO_GRANULARITY - seniorTR)), MathUpgradeable.Rounding.Up)
                .mulDiv(ONE, targetSubscriptionRatio, MathUpgradeable.Rounding.Up);

```

Reply:
However the current version seems easier to think about

```
juniorTR = (TRANCHE_RATIO_GRANULARITY - seniorTR)

# expected vault tvl 
vaultTVL' = perpTVL
.mulDiv(juniorTR, seniorTR, MathUpgradeable.Rounding.Up)
.mulDiv(targetSubscriptionRatio, ONE, MathUpgradeable.Rounding.Up)

# deviation
deviation = vaultTVL.mulDiv(ONE, vaultTVL')
```

Reply:
I'm normally all about self-documenting code, but I think we can make exceptions in cases where there's gas optimization to be gained and there's provable equivalence.

If we formulate it as:
DR = (vaultTVL * seniorTR) / (perpTVL * juniorTR * targetSR)

then I think we only need one mulDiv? (I'm omitting the ONEs for fixing decimals, but you get the idea)

This also matches to documentation we already have at the top of the file pretty closely

Reply:
Yea we can assume tvl*tr will be safe of overflow (tr <= 1000)

Updated to
```
uint256 juniorTR = TRANCHE_RATIO_GRANULARITY - seniorTR;
return (vaultTVL * seniorTR).mulDiv(ONE, (perpTVL * juniorTR)).mulDiv(ONE, targetSubscriptionRatio);
```